### PR TITLE
TRM-28694: index experimental evidence

### PR DESCRIPTION
**Pull request by LeonardoGonzales**

### CODE REVIEW

**Comment by jieluo321** [comment]
Both hasExperimentalEvidence function in UniProtEntryCommentsConverter need to consider evidence is empty for swissprot. Some old trembl entries may have annotations without evidence. I wonder if you need check if the entry is trembl or swissprot.

There are quite a few hardcoded field name inside the code such as "cc_ap_apu" , i wonder if you extract it as constant

**Comment by jieluo321** [comment]
For UniProtEntryFeatureConverter, the same issue for if evidence
  if (evidences.contains("experimental")) {

File: spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryFeatureConverter.java [hunk 1]
```diff
@@ -90,6 +80,18 @@ void convertFeature(List<UniProtKBFeature> features, UniProtDocument document) {
             Collection<String> evidenceList =
                     document.featureEvidenceMap.computeIfAbsent(evField, k -> new HashSet<>());
             evidenceList.addAll(evidences);
+
+            document.featuresMap
+                    .computeIfAbsent(field, k -> new HashSet<>())
+                    .addAll(featureValueList);
+            document.content.addAll(featureValueList);
+
+            if (evidences.contains("experimental")) {
```
Comment (ahmadshadab) [inline spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryFeatureConverter.java:89]: we can move this code to use same method as other converter

File: spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryCommentsConverter.java [hunk 2]
```diff
@@ -58,30 +58,25 @@ class UniProtEntryCommentsConverter implements Serializable {
...
-                String experimentalField = field + EXPERIMENTAL;
-                //  TODO the name should be plural
-                Collection<String> experimentalComment =
+                String experimentalField = commentField + EXPERIMENTAL;
+                Collection<String> experimentalComments =
                         document.commentMap.computeIfAbsent(
                                 experimentalField, k -> new ArrayList<>());
-                experimentalComment.add(commentVal);
+                experimentalComments.add(commentVal);
             }
```
Comment (ahmadshadab) [inline spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryCommentsConverter.java:81]: Could we replace this code block with  addExperimentalEvidence call?

**Review by ahmadshadab — APPROVED** [review]

File: spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryCommentsConverter.java (hunk 2 above)
Comment (LeonardoGonzales) [reply spark-indexer/src/main/java/org/uniprot/store/spark/indexer/uniprot/converter/UniProtEntryCommentsConverter.java:81]: The method created uses Set, and this part uses a List. this is the reason I did not add for this

**Pull request merged**